test/blockchain/pop/pop_vbk_block_tree_test.cpp

```diff
@@ -41,9 +100,121 @@ struct VbkBlockTreeTestFixture : ::testing::Test {
     vbk_repo = std::make_shared<BlockRepositoryInmem<BlockIndex<VbkBlock>>>();
     vbk_params = std::make_shared<VbkChainParamsRegTest>();
 
-    vbkTest =
-        std::make_shared<VbkBlockTreeTest>(*btcTree, vbk_repo, vbk_params);
+    endorsment_repo = std::make_shared<EndorsementsRepositoryInmem>();
+
+    btc_miner = std::make_shared<Miner<BtcBlock, BtcChainParams>>(btc_params);
+    vbk_miner = std::make_shared<Miner<VbkBlock, VbkChainParams>>(vbk_params);
+
+    vbkTest = std::make_shared<VbkBlockTreeTest>(
+        *btcTree, endorsment_repo, vbk_repo, vbk_params);
+
+    mock_miner = std::make_shared<MockMiner>();
+
+    setUpChains();
   }
 };
 
-TEST_F(VbkBlockTreeTestFixture, getProtoKeystoneContext_test) {}
+TEST_F(VbkBlockTreeTestFixture, getProtoKeystoneContext_test) {
+  uint32_t numVbkBlocks = 200;
+
+  mock_miner->mineVbkBlocks(numVbkBlocks, state);
+  ASSERT_TRUE(state.IsValid());
+
+  // initially we have to endorse the genesis block to update the vbkTest chain
+  // with the 200 context blocks from the mock_miner state
+  endorseVtbBlock(0);
+
+  ASSERT_TRUE(uint32_t(vbkTest->getBestChain().size() - 2) == numVbkBlocks);
```

Comment:
Why vbkTest chain is 2 blocks longer than mined in previous step?

Reply:
fixed